cmd/operator-sdk/generate/packagemanifests/cmd.go

```diff
@@ -0,0 +1,161 @@
+			if len(args) != 0 {
+				return fmt.Errorf("command %s doesn't accept any arguments", cmd.CommandPath())
+			}
+
+			// Check if the user has any specific preference to enable/disable
+			// interactive prompts. Default behaviour is to disable the prompt
+			// unless a base package does not exist.
+			if cmd.Flags().Changed("interactive") {
+				if c.interactive {
+					c.interactiveLevel = projutil.InteractiveOnAll
+				} else {
+					c.interactiveLevel = projutil.InteractiveHardOff
+				}
+			}
+
+			// Generate kustomize bases and manifests by default if no flags are set
+			// so the default behavior is "do everything".
+			fs := cmd.Flags()
+			if !fs.Changed("kustomize") && !fs.Changed("manifests") {
+				c.kustomize = true
+				c.manifests = true
+			}
+
+			cfg, err := kbutil.ReadConfig()
+			if err != nil {
+				log.Fatal(fmt.Errorf("error reading configuration: %v", err))
+			}
+			c.setCommonDefaults(cfg)
+
+			if c.kustomize {
+				if err = c.runKustomize(cfg); err != nil {
+					log.Fatalf("Error generating package bases: %v", err)
+				}
+			}
+			if c.manifests {
+				if err = c.validateManifests(); err != nil {
+					return fmt.Errorf("invalid command options: %v", err)
+				}
+				if err = c.runManifests(cfg); err != nil {
+					log.Fatalf("Error generating package manifests: %v", err)
+				}
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().BoolVar(&c.kustomize, "kustomize", false, "Generate kustomize bases")
+	cmd.Flags().BoolVar(&c.manifests, "manifests", false, "Generate package manifests")
+	cmd.Flags().BoolVar(&c.stdout, "stdout", false, "Write package to stdout")
+
+	c.addCommonFlagsTo(cmd.Flags())
+
+	return cmd
+}
+
+func (c *packagemanifestsCmd) addCommonFlagsTo(fs *pflag.FlagSet) {
+	fs.StringVar(&c.operatorName, "operator-name", "", "Name of the packaged operator")
+	fs.StringVarP(&c.version, "version", "v", "", "Semantic version of the packaged operator")
+	fs.StringVar(&c.inputDir, "input-dir", "", "Directory to read existing package manifests from. "+
+		"This directory is the parent of individual versioned package directories, and different from --deploy-dir")
+	fs.StringVar(&c.outputDir, "output-dir", "", "Directory in which to write package manifests")
+	fs.StringVar(&c.deployDir, "deploy-dir", "", "Root directory for operator manifests such as "+
+		"Deployments and RBAC, ex. 'deploy'. This directory is different from that passed to --input-dir")
+	fs.StringVar(&c.apisDir, "apis-dir", "", "Root directory for API type defintions")
+	fs.StringVar(&c.crdsDir, "crds-dir", "", "Root directory for CustomResoureDefinition manifests")
+	fs.StringVar(&c.channelName, "channel", "", "Channel name for the generated package")
+	fs.BoolVar(&c.isDefaultChannel, "default-channel", false, "Use the channel passed to --channel "+
+		"as the package manifest file's default channel")
+	fs.BoolVar(&c.updateCRDs, "update-crds", false, "Update CustomResoureDefinition manifests in this package. "+
```

Comment:
So `--update-crds=false` as the default implies the opposite of the help text `This will occur automatically when a new package is generated`.

Is it possible to simplify this by just having `--update-crds=true` as the default. 

And that simplifies our internal logic at [L222](https://github.com/operator-framework/operator-sdk/pull/3096/files#diff-e634c72f82054e9a0b9ed799ffa3e86bR222) as just:
```Go
if c.updateCRDs {
		var objs []interface{}
...
```

without having to do [L181-L183](https://github.com/operator-framework/operator-sdk/pull/3096/files#diff-e634c72f82054e9a0b9ed799ffa3e86bR181):
```Go
// When generating a new package, CRDs should be written unless --update-crds has been explicitly set to false.
updateCRDsOff := c.updateCRDsFlag.Changed && !c.updateCRDs
writeNewPackageCRDs := !updateCRDsOff && genutil.IsNotExist(packageDir)
```

Reply:
The UX gets a little weird here because you can update older package manifests versions of your operator, but you may not necessarily want to update CRDs for that version. That was the original thought at least.

Now that I think about this, I can imagine that developers tend to work on the latest package manifests version in master then check out the related release branch if they need to work on a prior version, in which the older version is latest. Given that premise I am ok with defaulting to `--update-crds=true`. Thoughts @joelanford?